test/hotspot/gtest/utilities/test_ostream.cpp

```diff
@@ -123,6 +124,257 @@ TEST_VM(ostream, bufferedStream_dynamic_large) {
 
 */
 
+// Test helper for do_vsnprintf
+class outputStream::TestSupport : AllStatic {
 
+  // Shared constants and variables for all subtests.
+  static const size_t buflen = 11;
+  static char buffer[buflen];
+  static const size_t max_len = buflen - 1;
+  static size_t result_len;
+  static const char* result;
 
+  static void reset() {
+    result_len = 0;
+    result = nullptr;
+    buffer[0] = '\0';
+  }
+
+  static const char* test(char* buf, size_t len, bool add_cr,
+                          size_t& rlen, const char* format, ...) {
+    va_list ap;
+    va_start(ap, format);
+    const char* res = do_vsnprintf(buf, len, format, ap, add_cr, rlen);
+    va_end(ap);
+    return res;
+  }
+
+ public:
+  // Case set 1: constant string with no format specifiers
+  static void test_constant_string() {
+    reset();
+    // Case 1-1: no cr, no truncation, excess capacity
+    {
+      const char* str = "012345678";
+      size_t initial_len = strlen(str);
+      ASSERT_TRUE(initial_len < max_len);
+      result = test(&buffer[0], buflen, false, result_len, str);
```

Comment:
I must have forgotten to add this comment before but here it is again:
Is there a reason you chose to use `&buffer[0]` rather than simple `buffer`?

Reply:
At some point when writing the test code it wouldn't compile (perhaps I had a misplaced const at the time?) but now it does - fixed.